controllers/argocd/secret/secret.go

```diff
@@ -17,7 +39,494 @@ type SecretReconciler struct {
 }
 
 func (sr *SecretReconciler) Reconcile() error {
+	var reconciliationErrors []error
+	sr.Logger = ctrl.Log.WithName(SecretsControllerName).WithValues("instance", sr.Instance.Name, "instance-namespace", sr.Instance.Namespace)
+
+	if err := sr.reconcileCredentialsSecret(); err != nil {
+		reconciliationErrors = append(reconciliationErrors, err)
+	}
+
+	if err := sr.reconcileClusterPermissionsSecret(); err != nil {
+		reconciliationErrors = append(reconciliationErrors, err)
+	}
+
+	if err := sr.reconcileCASecret(); err != nil {
+		return err
+	}
+
+	if err := sr.reconcileTLSSecret(); err != nil {
+		return err
+	}
+
+	if err := sr.reconcileArgoCDSecret(); err != nil {
+		return err
+	}
+
+	// TO DO : skipping grafana for now - it should either be in its own controller
+	// or preferably skipped altogether by the operator
+
+	return amerr.NewAggregate(reconciliationErrors)
+}
+
+// reconcileArgoCDSecret creates and updates the "argocd-secret" secret for the given
+// Argo CD instance. It requires the credentials secret, CA secret and TLS secret
+// to be present on the cluster already
+func (sr *SecretReconciler) reconcileArgoCDSecret() error {
+	credSecretName := util.NameWithSuffix(sr.Instance.Name, DefaultClusterCredentialsSuffix)
+	credsSecret, err := workloads.GetSecret(credSecretName, sr.Instance.Namespace, sr.Client)
+	if err != nil {
+		sr.Logger.Error(err, "reconcileArgoCDSecret: failed to retrieve secret", "name", credSecretName, "namespace", sr.Instance.Namespace)
```

Comment:
Do we need to add the "one or more mutations could not be applied" log here as well?